shotover-proxy/tests/kafka_int_tests/mod.rs

```diff
@@ -316,6 +334,20 @@ async fn cluster_sasl_plain_multi_shotover(#[case] driver: KafkaDriver) {
         KafkaConnectionBuilder::new(driver, "127.0.0.1:9192").use_sasl_plain("user", "password");
     test_cases::standard_test_suite(connection_builder).await;
 
+    // Test invalid credentials
+    // We perform the regular test suite first in an attempt to catch a scenario
+    // where valid credentials are leaked into subsequent connections.
+    let connection_builder = KafkaConnectionBuilder::new(driver, "127.0.0.1:9192")
+        .use_sasl_plain("user", "not_the_password");
+    assert_eq!(
+        connection_builder.assert_admin_error().await.to_string(),
+        match driver {
+            #[cfg(feature = "rdkafka-driver-tests")]
```

Comment:
Seems every test in this module has this feature guard? Might be able to move it up a level - not for this PR though.

Reply:
This feature guard is for the cpp driver, the test cases that use java driver can still be run when its disabled.
Now that you've brought this up, I'm thinking rdkafka-driver-tests should be renamed to `kafka-cpp-driver-tests`, since its not obvious that rdkafka is the name of the cpp driver for kafka. I'll do this in another PR.